CRM/Utils/Date.php

```diff
@@ -873,8 +873,11 @@ public static function getRange($startDate, $endDate) {
    */
   public static function getFromTo($relative, $from, $to, $fromTime = NULL, $toTime = '235959') {
     if ($relative) {
-      list($term, $unit) = explode('.', $relative, 2);
-      $dateRange = self::relativeToAbsolute($term, $unit);
+      $dateRange = CRM_Utils_Hook::relativeDate($relative);
+      if (!is_array($dateRange) || empty($dateRange['from'])) {
```

Comment:
I think it needs to be (empty($dateRange['from']) && empty($dateRange['to'])) - it's legit for only one to be set